root/url/utility/isUrlValid.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * @flow strict
+ * Copyright (C) 2021 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+export default function isUrlValid(url: string): boolean {
+  try {
+    new URL(url);
```

Comment:
Well, when I wrote that in 2014 the `URL` API was brand new (if supported at all outside of Chrome) and we didn't have core-js polyfills or anything back then.

One difference that probably exists, though, is that something like `new URL('/haha')` is valid, whereas the current function requires a fully-specified URL.  This seems like something that should be looked into, if it matters.

Reply:
That's not quite true - https://developer.mozilla.org/en-US/docs/Web/API/URL/URL "If url is a relative URL, base is required, and will be used as the base URL". Otherwise, it errors anyway.